Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp

```diff
@@ -1327,24 +1328,34 @@ void GStreamerMediaEndpoint::createSessionDescriptionSucceeded(GUniquePtr<GstWeb
             return;
 
         GUniquePtr<char> sdp(gst_sdp_message_as_text(description->sdp));
-        if (m_isInitiator)
-            m_peerConnectionBackend.createOfferSucceeded(String::fromLatin1(sdp.get()));
-        else
-            m_peerConnectionBackend.createAnswerSucceeded(String::fromLatin1(sdp.get()));
+        auto sdpString = String::fromLatin1(sdp.get());
+        if (description->type == GST_WEBRTC_SDP_TYPE_OFFER) {
+            m_peerConnectionBackend.createOfferSucceeded(WTFMove(sdpString));
+            return;
+        }
+
+        if (description->type == GST_WEBRTC_SDP_TYPE_ANSWER) {
+            m_peerConnectionBackend.createAnswerSucceeded(WTFMove(sdpString));
+            return;
+        }
+
+        GST_WARNING_OBJECT(m_pipeline.get(), "Unsupported SDP type: %s", gst_webrtc_sdp_type_to_string(description->type));
     });
 }
 
-void GStreamerMediaEndpoint::createSessionDescriptionFailed(GUniquePtr<GError>&& error)
+void GStreamerMediaEndpoint::createSessionDescriptionFailed(RTCSdpType sdpType, GUniquePtr<GError>&& error)
 {
-    callOnMainThread([protectedThis = Ref(*this), this, error = WTFMove(error)] {
+    callOnMainThread([protectedThis = Ref(*this), this, sdpType, error = WTFMove(error)] {
         if (isStopped())
             return;
 
         auto exc = Exception { OperationError, error ? String::fromUTF8(error->message) : "Unknown Error"_s };
-        if (m_isInitiator)
+        if (sdpType == RTCSdpType::Offer)
             m_peerConnectionBackend.createOfferFailed(WTFMove(exc));
-        else
+        else {
+            ASSERT(sdpType == RTCSdpType::Answer);
```

Comment:
Can this ever be otherwise?

Reply:
Nop

Reply:
Do you want me to remove this ASSERT?

Reply:
If it can't be otherwise because this enum has only two cases, the only case where this could hit is corruption. If you don't suspect this is a plausible case, I would remove it.